test/e2e/tests/api-usage/account-tracker-api-usage.spec.ts

```diff
@@ -0,0 +1,173 @@
+import { strict as assert } from 'assert';
+import { MockedEndpoint } from 'mockttp';
+import { JsonRpcRequest } from '@metamask/utils';
+import {
+  withFixtures,
+  defaultGanacheOptions,
+  unlockWallet,
+} from '../../helpers';
+import FixtureBuilder from '../../fixture-builder';
+import { Driver } from '../../webdriver/driver';
+import { Mockttp } from '../../mock-e2e';
+
+async function mockInfura(mockServer: Mockttp): Promise<MockedEndpoint[]> {
+  const blockNumber = { value: 0 };
+
+  return [
+    await mockServer
+      .forPost(/infura/u)
+      .withJsonBodyIncluding({ method: 'eth_blockNumber' })
+      .times(50)
+      .thenCallback(() => {
+        blockNumber.value += 1;
+
+        return {
+          statusCode: 200,
+          json: {
+            jsonrpc: '2.0',
+            id: '1111111111111111',
+            result: blockNumber.value.toString(16),
+          },
+        };
+      }),
+    await mockServer.forPost(/infura/u).thenCallback(() => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: '1111111111111111',
+          result: '0x1',
+        },
+      };
+    }),
+    await mockServer.forGet(/infura/u).thenCallback(() => {
+      return {
+        statusCode: 200,
+        json: {
+          jsonrpc: '2.0',
+          id: '1111111111111111',
+          result: '0x0',
+        },
+      };
+    }),
+  ];
+}
+
+async function getAllInfuraJsonRpcRequests(
+  mockedEndpoint: MockedEndpoint[],
+): Promise<JsonRpcRequest[]> {
+  const allInfuraJsonRpcRequests: JsonRpcRequest[] = [];
+  let seenRequests;
+  let seenProviderRequests;
+
+  for (const m of mockedEndpoint) {
+    seenRequests = await m.getSeenRequests();
+    seenProviderRequests = seenRequests.filter((request) =>
+      request.url.match('infura'),
+    );
+
+    for (const r of seenProviderRequests) {
+      const json = await r.body.getJson();
+      if (json !== undefined) {
+        allInfuraJsonRpcRequests.push(json);
+      }
+    }
+  }
+
+  return allInfuraJsonRpcRequests;
+}
+
+describe('Account Tracker API Usage', function () {
+  it('should not make eth_call or eth_getBalance requests before the UI is opened and should make those requests after the UI is opened', async function () {
```

Comment:
I see that the account tracker is also responsible from initiating `eth_getBlockByNumber` calls. Should we also include them here?

https://github.com/user-attachments/assets/871e7182-0221-40c3-a80c-10f68ad8b6b9

Reply:
good suggestion, added here ac50f666ee

Reply:
thank you Dan, great refactor! Could it be that we are missing it [here](https://github.com/MetaMask/metamask-extension/commit/ac50f666eebae7403308e438d4d4598d0a762778#diff-174d8183603b0004a7ff4acb5a21374ea2ad88e63ab433090783ef41db9c35e7R86) and in the test description (not a blocker though)?

Reply:
No, in that case there is an eth_getBlockbyNumber call before the UI opens, for other reasons. That should be addressed separately.

I will create a ticket for that and add a code comment and update the test description